File: study/자료구조withPython/josephus/josephus.py
import linked_queue

def remover(que, k):
    for i in range(k - 1):
        que.enqueue(que.dequeue())
    return que.dequeue()

n = int(input())
k = int(input())
que = linked_queue.LinkedQueue()

for i in range(n):
    que.enqueue(i + 1)

for i in range(n):
    print(remover(que, k), end=" ")

File: 2020-1/자료구조/Report04/Josephus.cpp
#define _CRT_SECURE_NO_WARNINGS
#include <cstdio>
#include <cstdlib>
#include "Josephus.h"

class Remover {
	Josephus josep;
	int n;
	int k;

public:
	Remover(int _n, int _k) : josep(), n(_n), k(_k) {
		for (int i = 0; i < n; i++)
			josep.enqueue(i + 1);
	}

	int getNextVal() {
		for (int i = 0; i < k - 1; i++) {
			josep.enqueue(josep.dequeue());
		}
		return josep.dequeue();
	}
};

int main() {
	int n = 0;
	int k = 0;

	scanf("%d", &n);
	scanf("%d", &k);

	Remover obj(n, k);

	printf("\n<");
	for (int i = 0; i < n; i++)
		printf("%d%s", obj.getNextVal(), i==n-1 ? ">" : ", ");
	printf("\n");
}